File: src/main/java/com/refinedmods/refinedstorage/screen/widget/sidebutton/AccessTypeSideButton.java
package com.refinedmods.refinedstorage.screen.widget.sidebutton;

import com.refinedmods.refinedstorage.api.storage.AccessType;
import com.refinedmods.refinedstorage.blockentity.data.BlockEntitySynchronizationManager;
import com.refinedmods.refinedstorage.blockentity.data.BlockEntitySynchronizationParameter;
import com.refinedmods.refinedstorage.screen.BaseScreen;
import com.refinedmods.refinedstorage.util.AccessTypeUtils;
import net.minecraft.ChatFormatting;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.client.resources.language.I18n;

public class AccessTypeSideButton extends SideButton {
    private final BlockEntitySynchronizationParameter<AccessType, ?> parameter;

    public AccessTypeSideButton(BaseScreen<?> screen, BlockEntitySynchronizationParameter<AccessType, ?> parameter) {
        super(screen);

        this.parameter = parameter;
    }

    @Override
    protected void renderButtonIcon(GuiGraphics graphics, int x, int y) {
        graphics.blit(BaseScreen.ICONS_TEXTURE, x, y, 16 * parameter.getValue().getId(), 240, 16, 16);
    }

    @Override
    protected String getSideButtonTooltip() {
        return I18n.get("sidebutton.refinedstorage.access_type") + "\n" + ChatFormatting.GRAY + I18n.get("sidebutton.refinedstorage.access_type." + parameter.getValue().getId());
    }

    @Override
    public void onPress() {
        BlockEntitySynchronizationManager.setParameter(parameter, AccessTypeUtils.getAccessType(parameter.getValue().getId() + 1));
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/widget/sidebutton/AccessTypeSideButton.kt
package com.refinedmods.refinedstorage.screen.widget.sidebutton

import com.mojang.blaze3d.matrix.MatrixStack
import com.refinedmods.refinedstorage.api.storage.AccessType
import com.refinedmods.refinedstorage.screen.BaseScreen
import com.refinedmods.refinedstorage.tile.data.TileDataManager
import com.refinedmods.refinedstorage.tile.data.TileDataParameter
import com.refinedmods.refinedstorage.util.AccessTypeUtils
import net.minecraft.client.resources.I18n
import net.minecraft.util.text.TextFormatting

class AccessTypeSideButton(screen: BaseScreen<*>, private val parameter: TileDataParameter<AccessType?, *>) : SideButton(screen) {
    override fun renderButtonIcon(matrixStack: MatrixStack?, x: Int, y: Int) {
        screen.blit(matrixStack, x, y, 16 * parameter.value!!.getId(), 240, 16, 16)
    }

    override fun getTooltip(): String {
        return I18n.format("sidebutton.refinedstorage.access_type").toString() + "\n" + TextFormatting.GRAY + I18n.format("sidebutton.refinedstorage.access_type." + parameter.value!!.getId())
    }

    fun onPress() {
        TileDataManager.setParameter(parameter, AccessTypeUtils.getAccessType(parameter.value!!.getId() + 1))
    }
}